### PR TITLE
Replace interlocked increment/decrement on QAtomicInteger for SharedPtr

### PR DESCRIPTION
**Pull request by nk0leg**

### CODE REVIEW

**Comment by nk0leg** [comment]
@anosach-luxoft , @malirod, @pestOO  please review

**Comment by nk0leg** [comment]
Temporarily suspended because of the regular crashes during deleting QAtomicInt

**Comment by anosach-luxoft** [comment]
@nk0leg what about using only `++` and `--` operators?

**Comment by malirod** [comment]
@anosach-luxoft, I was thinking about these `++` and `--`. In the Qt impl of the shared ptr, they use `ref``deref`. Since we don't need the old value (result of`++`and`--`) I think we can keep`ref`and`deref`. And in one place we really need`--` since we analyse the old value. So I propose to leave the usage as it is.

File: src/components/include/utils/shared_ptr.h [hunk 1]
```diff
@@ -38,8 +38,11 @@
 #include <stdint.h>
 
 #include "utils/macro.h"
+#ifdef QT_PORT
+#include <QAtomicPointer>
```
Comment (anosach-luxoft) [inline src/components/include/utils/shared_ptr.h]: `#include <QAtomicInt>`

Comment (nk0leg) [inline src/components/include/utils/shared_ptr.h]: see 38ee329

**Pull request merged**